From: David Hildenbrand <david@redhat.com>
Subject: fixup: mm/mprotect: drop pgprot_t parameter from change_protection()
Date: Sat, 24 Dec 2022 15:01:18 +0100

PAGE_NONE might not be defined without CONFIG_NUMA_BALANCING

Link: https://lkml.kernel.org/r/5084ff1c-ebb3-f918-6a60-bacabf550a88@redhat.com
Signed-off-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---


--- a/mm/mprotect.c~mm-mprotect-drop-pgprot_t-parameter-from-change_protection-fix
+++ a/mm/mprotect.c
@@ -514,6 +514,7 @@ unsigned long change_protection(struct m
 
 	BUG_ON((cp_flags & MM_CP_UFFD_WP_ALL) == MM_CP_UFFD_WP_ALL);
 
+#ifdef CONFIG_NUMA_BALANCING
 	/*
 	 * Ordinary protection updates (mprotect, uffd-wp, softdirty tracking)
 	 * are expected to reflect their requirements via VMA flags such that
@@ -521,6 +522,9 @@ unsigned long change_protection(struct m
 	 */
 	if (cp_flags & MM_CP_PROT_NUMA)
 		newprot = PAGE_NONE;
+#else
+	WARN_ON_ONCE(cp_flags & MM_CP_PROT_NUMA);
+#endif
 
 	if (is_vm_hugetlb_page(vma))
 		pages = hugetlb_change_protection(vma, start, end, newprot,
_
